internal/driver/fetch.go

```diff
@@ -589,7 +589,7 @@ func adjustURL(source string, duration, timeout time.Duration) (string, time.Dur
 		// url.Parse treats "hostname" as the scheme.
 		u, err = url.Parse("http://" + source)
 	}
-	if err != nil || u.Host == "" {
+	if err != nil || u.Host == "" || isFile(source) {
```

Comment:
I would put this check in the `fetch()` function, before we even calling `adjustURL()`. This will colocate it with other details of handling files vs URLs making it easier to read the code.

Can you also see if you can add a test that fails before the change and passes with the fix?

Reply:
Of course. Thanks

Reply:
Modified, please review again and give suggestions